geode-redis/src/integrationTest/java/org/apache/geode/redis/general/PersistIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+    String value = "value";
+
+    jedis.sadd(key, value);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithSortedSetValue() {
+    String key = "key";
+    double score = 2.0;
+    String member = "member";
+
+    jedis.zadd(key, score, member);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithHashValue() {
+    String key = "key";
+    String field = "field";
+    String value = "value";
+
+    jedis.hset(key, field, value);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithGeoValue() {
+    String key = "sicily";
+    double latitude = 13.361389;
+    double longitude = 38.115556;
+    String member = "Palermo Catina";
+
+    jedis.geoadd(key, latitude, longitude, member);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithHyperLogLogValue() {
+    String key = "crawled:127.0.0.2";
+    String value = "www.insideTheHouse.com";
+
+    jedis.pfadd(key, value);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithListValue() {
+    String key = "list";
+    String value = "value";
+
+    jedis.lpush(key, value);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKey_givenKeyWithBitMapValue() {
+    String key = "key";
+    long offset = 1L;
+    String value = "0";
+
+    jedis.setbit(key, offset, value);
+    jedis.expire(key, 20);
+
+    assertThat(jedis.persist(key)).isEqualTo(1L);
+    assertThat(jedis.ttl(key)).isEqualTo(-1L);
+  }
+
+  @Test
+  public void shouldPersistKeysConcurrently() throws InterruptedException {
+    doABunchOfSetEXsWithBlockingQueue(jedis);
+
+    AtomicLong persistedFromThread1 = new AtomicLong(0);
+    AtomicLong persistedFromThread2 = new AtomicLong(0);
+
+    Runnable runnable1 = () -> doABunchOfPersistsWithBlockingQueue(persistedFromThread1, jedis);
+    Runnable runnable2 = () -> doABunchOfPersistsWithBlockingQueue(persistedFromThread2, jedis2);
+
+    Thread thread1 = new Thread(runnable1);
+    Thread thread2 = new Thread(runnable2);
+
+    thread1.start();
+    thread2.start();
+    thread1.join();
+    thread2.join();
+
+    assertThat(persistedFromThread1.get() + persistedFromThread2.get()).isEqualTo(ITERATION_COUNT);
+  }
+
+  private void doABunchOfSetEXsWithBlockingQueue(Jedis jedis) {
```

Comment:
where's the blocking queue here?  this name may be a relic, and needs to be updated?

"do a bunch of" is odd.  maybe the method is just expireKeys, and  below one is persistKeys

Reply:
It was a relic! We forgot to rename the method names.  Thank you!  As far as "do a bunch of", we were following a naming convention from a few other test suites.  I changed it though.  Maybe we should discuss as a team or rename the other methods as well?